RecoLocalCalo/HGCalRecProducers/interface/HGCalImagingAlgo.h

```diff
@@ -177,19 +84,44 @@ void reset(){
 void computeThreshold();
 
 //getDensity
- Density getDensity();
+ Density getDensity() override;
+
+static void fillPSetDescription(edm::ParameterSetDescription& iDesc) {
```

Comment:
This PSet description is the same as the one above for HGCalCLUEAlgo (a part the different numerical values for "deltac"): shouldn't this method be included in the base class, instead?

Reply:
@perrotta this is true for the moment. There is work ongoing to try and factor few parameters out depending on which part of the detector you are in. And, for sure, the scintillator part will have to be further investigated. Coupling the PSet together means to verify, everytime we change them or add anything, that the old algorithm still works as expected, which is something I do not think is useful. I therefore propose to decouple them completely, if you agree.

Reply:
Sorry, but I'm missing the point.
Even if you have the parameter set description in common, the actual configurations will be different in the two cases. And if one algo needs some extra parameter not needed by the other algo, that parameter can be added to the pset description of one algo and not to the pset description of the other one.

Reply:
The shared and commonly used parameters would have to be validated twice, and, very likely, the working points will be different for the different algorithm. Also, if we introduce new parameters we will likely remove other but, if the 2 are coupled, we could not do that and will end up having stale parameters in one of the 2.